SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -0,0 +1,568 @@
+	const Schema_Object* SchemaObject = Schema_GetComponentDataFields(ComponentData.schema_type);
+	uint32 InitiallyPresent = Schema_GetUint32(SchemaObject, SpatialGDK::RPCRingBufferUtils::GetInitiallyPresentMulticastRPCsCountFieldId());
+	bool bTestPassed = (InitiallyPresent == 5);
+
+	TestTrue("Entity creation multi-cast test returned expected results", bTestPassed);
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_push_server_reliable_rpcs_to_the_service_THEN_extracted_payloads_match_pushed_payloads)
+{
+	USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+
+	Worker_Authority ClientAuth = GetClientAuthorityFromRPCEndpointType(SERVER_AUTH);
+	Worker_Authority ServerAuth = GetServerAuthorityFromRPCEndpointType(SERVER_AUTH);
+
+	Schema_ComponentData* ClientComponentData = Schema_CreateComponentData();
+	Schema_Object* ClientSchemaObject = Schema_GetComponentDataFields(ClientComponentData);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 1, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 2, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 3, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 4, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 5, SimplePayload);
+
+	AddEntityComponentToStaticComponentView(*StaticComponentView,
+		SpatialGDK::EntityComponentId(RPCTestEntityId_1, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID),
+		ClientComponentData,
+		ClientAuth);
+
+	AddEntityComponentToStaticComponentView(*StaticComponentView,
+		SpatialGDK::EntityComponentId(RPCTestEntityId_1, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID),
+		ServerAuth);
+
+	bool bAttemptedRPCExtraction = false;
+	bool bPayloadsMatch = true;
+	ExtractRPCDelegate RPCDelegate = ExtractRPCDelegate::CreateLambda([&bAttemptedRPCExtraction, &bPayloadsMatch](Worker_EntityId EntityId, ERPCType RPCType, const SpatialGDK::RPCPayload& Payload) {
+		bAttemptedRPCExtraction = true;
+		bPayloadsMatch &= CompareRPCPayload(Payload, SimplePayload);
+		bPayloadsMatch &= EntityId == RPCTestEntityId_1;
+		return true;
+	});
+
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, SERVER_AUTH, RPCDelegate, StaticComponentView);
+	RPCService.ExtractRPCsForEntity(RPCTestEntityId_1, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID);
+
+	bool bTestPassed = bAttemptedRPCExtraction && bPayloadsMatch;
+	TestTrue("Extracted RPCs match expected payloads", bTestPassed);
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_return_false_from_extract_callback_THEN_some_rpcs_persist_on_component)
```

Comment:
I think the authority is less important than you receiving an rpc for the precondition.